### PR TITLE
For login, replace username by email, add template for password reset 

### PR DESCRIPTION
**Pull request by katjad**
Authentication via email :)

### CODE REVIEW

File: bookx/settings.py [hunk 1]
```diff
@@ -137,7 +137,8 @@
 MEDIA_ROOT = os.path.join(BASE_DIR, 'media')
 
 AUTH_USER_MODEL = 'users.CustomUser'
-
+AUTHENTICATION_BACKENDS = ['users.backends.EmailBackend','django.contrib.auth.backends.ModelBackend']
```
Comment (Geekfish) [inline bookx/settings.py:140]: Do you still want to keep the `ModelBackend` as well?

**Review by Geekfish — COMMENTED** [review]
Very nice! ⭐️ 

In the past I've done this in a similar way as you did here. I think now django lets you specify a different "username" field, via the [USERNAME_FIELD](https://docs.djangoproject.com/en/2.1/topics/auth/customizing/#django.contrib.auth.models.CustomUser.USERNAME_FIELD) setting.
Could setting this to `"email"` remove the need to override some of these things?
Might be a thing to explore!

Happy to merge this for now if you are!

**Comment by katjad** [comment]
Yes I saw about USERNAME_FIELD but I thought that in that case you would completely replace the username by email, and if you wanted a username have to add it on again? But I might be wrong about that. 

In any case, in the end the alternate authentication backend seemed the easiest and I was surprised I didn't have to change any more than the template (not done very eleganty I have to admit!). So I'd be happy to have it merged in as it is :) Thanks!

File: bookx/settings.py (hunk 1 above)
Comment (katjad) [reply bookx/settings.py:140]: Just realised I hadn't answered this yet. Mmh I don't really know :) Maybe just so I know where the original one is..

**Comment by Geekfish** [comment]
I think all that "USERNAME_FIELD" describes is "which field is the 'identifying' field for our users"?
You can still keep any other fields you like.

**Comment by katjad** [comment]
Oh I see! That's good to know. Thanks for merging. 

Been working on Dockerfile an docker-compose, hope another pull request incoming today :)

**Pull request merged**